### PR TITLE
Add "Adversarial GLUE" dataset to datasets library

### PR DESCRIPTION
**Pull request by jxmorris12**
Adds the Adversarial GLUE dataset: https://adversarialglue.github.io/

```python
>>> import datasets
>>> >>> datasets.load_dataset('adv_glue')
Using the latest cached version of the module from /home/jxm3/.cache/huggingface/modules/datasets_modules/datasets/adv_glue/26709a83facad2830d72d4419dd179c0be092f4ad3303ad0ebe815d0cdba5cb4 (last modified on Mon Mar  7 19:19:48 2022) since it couldn't be found locally at adv_glue., or remotely on the Hugging Face Hub.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/jxm3/random/datasets/src/datasets/load.py", line 1657, in load_dataset
    builder_instance = load_dataset_builder(
  File "/home/jxm3/random/datasets/src/datasets/load.py", line 1510, in load_dataset_builder
    builder_instance: DatasetBuilder = builder_cls(
  File "/home/jxm3/random/datasets/src/datasets/builder.py", line 1021, in __init__
    super().__init__(*args, **kwargs)
  File "/home/jxm3/random/datasets/src/datasets/builder.py", line 258, in __init__
    self.config, self.config_id = self._create_builder_config(
  File "/home/jxm3/random/datasets/src/datasets/builder.py", line 337, in _create_builder_config
    raise ValueError(
ValueError: Config name is missing.
Please pick one among the available configs: ['adv_sst2', 'adv_qqp', 'adv_mnli', 'adv_mnli_mismatched', 'adv_qnli', 'adv_rte']
Example of usage:
	`load_dataset('adv_glue', 'adv_sst2')`
>>> datasets.load_dataset('adv_glue', 'adv_sst2')['validation'][0]
Reusing dataset adv_glue (/home/jxm3/.cache/huggingface/datasets/adv_glue/adv_sst2/1.0.0/3719a903f606f2c96654d87b421bc01114c37084057cdccae65cd7bc24b10933)
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 604.11it/s]
{'sentence': "it 's an uneven treat that bores fun at the democratic exercise while also examining its significance for those who take part .", 'label': 1, 'idx': 0}
```

### CODE REVIEW

**Comment by jxmorris12** [comment]
@lhoestq can you review when you have some time?

File: datasets/adv_glue/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+---
+pretty_name: Adversarial GLUE
+languages:
+- en
+paperswithcode_id: adversarial-glue-a-multi-task-benchmark-for
```
Comment (lhoestq) [inline datasets/adv_glue/README.md:5]: Could you add the missing tags here ? feel free to use https://huggingface.co/spaces/huggingface/datasets-tagging

File: datasets/adv_glue/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+AdvGLUE deviates from the GLUE dataset, which has a base language of English.
+## Dataset Structure
+
+We show detailed information for up to 5 configurations of the dataset.
+
+### Data Instances
+
+#### default
+
+- **Size of downloaded dataset files:** 198 KB
```
Comment (lhoestq) [inline datasets/adv_glue/README.md:31]: Could you provide an actual example from the dataset here ? Feel free to use `load_dataset` and take the first one

File: datasets/adv_glue/README.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
+---
+pretty_name: Adversarial GLUE
+languages:
+- en
+paperswithcode_id: adversarial-glue-a-multi-task-benchmark-for
+---
+
+# Dataset Card for "adv_glue"
+
```
Comment (lhoestq) [inline datasets/adv_glue/README.md:9]: It looks like the table of content is missing here, feel free to copy it from the template here: https://github.com/huggingface/datasets/tree/master/templates

Note that every section should also appear in the dataset card, though it's ok if some of them are empty or simply contains `[More Information Needed]`

File: datasets/adv_glue/README.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+## Dataset Description
+
```
Comment (lhoestq) [inline datasets/adv_glue/README.md:11]:
```suggestion

- **Homepage:**  https://adversarialglue.github.io/
- **Repository:**
- **Paper:** [arXiv](https://arxiv.org/pdf/2111.02840.pdf)
- **Leaderboard:**
- **Point of Contact:**
```

**Review by lhoestq — COMMENTED** [review]
Hi ! Thanks for adding this one :) The dataset script looks all good and the dummy data as well. I added a few comments for the dataset card:

File: datasets/adv_glue/README.md (hunk 2 above)
Comment (jxmorris12) [reply datasets/adv_glue/README.md:36]: done!

File: datasets/adv_glue/README.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+- **Size of downloaded dataset files:** 198 KB
+- **Example**:
+```python
+>>> datasets.load_dataset('adv_glue', 'adv_sst2')['validation'][0]
+{'sentence': "it 's an uneven treat that bores fun at the democratic exercise while also examining its significance for those who take part .", 'label': 1, 'idx': 0}
+```
+
+### Data Fields
+
+The data fields are the same as in the GLUE dataset, which differ by task.
```
Comment (lhoestq) [inline datasets/adv_glue/README.md:79]: Feel free to copy/paster the fields from GLUE here

File: datasets/adv_glue/README.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
+---
+annotations_creators: []
+language_creators: []
+languages: ['en']
+licenses: []
+multilinguality: []
+pretty_name: Adversarial GLUE
+paperswithcode_id: adversarial-glue-a-multi-task-benchmark-for
+size_categories:
+- unknown
```
Comment (lhoestq) [inline datasets/adv_glue/README.md:10]:
```suggestion
- n<1K
```

**Review by lhoestq — COMMENTED** [review]
Cool ! I left some comments about the tags, I think we can merge once they're added

Thank you so much !

**Comment by lhoestq** [comment]
Feel free to also merge `master` into your branch to get the latest updates for the tests ;)

File: datasets/adv_glue/README.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+source_datasets: []
+task_categories: []
+task_ids: []
```
[the remark this answers is not in the document]
Comment (jxmorris12) [reply datasets/adv_glue/README.md]: yes, just added! The web app is really nice!

File: datasets/adv_glue/README.md (hunk 5 above)
Comment (jxmorris12) [reply datasets/adv_glue/README.md:87]: Done! Although most of them say "More Information Needed"

**Comment by jxmorris12** [comment]
thanks @lhoestq - just made all the updates you requested!

**Review by lhoestq — APPROVED** [review]
Thanks a lot ! merging now :)

**Pull request merged**